routers/web/repo/release.go

```diff
@@ -489,7 +489,7 @@ func NewReleasePost(ctx *context.Context) {
 			return
 		}
 	} else {
-		if !rel.IsTag {
+		if len(form.TagOnly) > 0 && rel.IsTag {
```

Comment:
Modern Golang uses `form.TagOnly != ""`

```suggestion
		if form.TagOnly != "" && rel.IsTag {
```